x-pack/plugins/lens/public/visualizations/partition/render_helpers.ts

```diff
@@ -6,15 +6,17 @@
  */
 
 import type { Datatable } from '@kbn/expressions-plugin/public';
+import { LegendStats } from '@kbn/visualizations-plugin/common/constants';
 import type { PieChartType, PieLayerState } from '../../../common/types';
 import { PartitionChartsMeta } from './partition_charts_meta';
 
-export const shouldShowValuesInLegend = (layer: PieLayerState, shape: PieChartType) => {
-  if ('showValues' in PartitionChartsMeta[shape]?.legend) {
-    return layer.showValuesInLegend ?? PartitionChartsMeta[shape]?.legend?.showValues ?? true;
+export const getLegendStats = (layer: PieLayerState, shape: PieChartType) => {
+  if ('defaultLegendStats' in PartitionChartsMeta[shape]?.legend) {
+    return (
+      layer.legendStats ??
+      PartitionChartsMeta[shape]?.legend?.defaultLegendStats ?? [LegendStats.values]
+    );
```

Comment:
This (and the previous code too) implies that the `PartitionChartsMeta[shape]` can be `undefined` in some cases, but that should not be possible as the `in` operator would throw in a `defaultLegendStats in undefined` and not report `false`.

I propose to make this clearer:

```suggestion
  if ('defaultLegendStats' in PartitionChartsMeta[shape].legend) {
    return (
      layer.legendStats ??
      PartitionChartsMeta[shape].legend.defaultLegendStats ?? 
      [LegendStats.values]
    );
```